### PR TITLE
Blood rites refactor

### PR DESCRIPTION
**Pull request by HMBGERDO**
## What Does This PR Do

Cleans up code, makes blood rites collect blood if you didnt click human or construct by default(prev you needed to click blood itself), allows you to collect blood in full darkness

## Why It's Good For The Game

blood rites not working in full darkness while all cult is wearing nightvision blindfold is stupid

`if`
`   if`
`        if`
`            if`
code is bad

## Testing

compiled, tested

## Changelog
:cl:
tweak: Now you can click any object with Blood Rites to collect blood
tweak: Applying Blood Rites on cultist cleans their blood on clothes
fix: Now Blood Rites blood collection works in full darkness
/:cl:

### CODE REVIEW

File: code/game/gamemodes/cult/blood_magic.dm [hunk 1]
```diff
@@ -735,148 +735,163 @@
...
-		for(var/obj/effect/decal/cleanable/trail_holder/TH in view(T, 2))
-			qdel(TH)
-		if(temp)
-			user.Beam(T, icon_state = "drainbeam", time = 15)
-			new /obj/effect/temp_visual/cult/sparks(get_turf(user))
-			playsound(T, 'sound/misc/enter_blood.ogg', 50)
-			temp = round(temp)
-			to_chat(user, "<span class='cultitalic'>Your blood rite has gained [temp] charge\s from blood sources around you!</span>")
-			uses += max(1, temp)
+	var/turf/T = isturf(target) ? target : get_turf(target)
```
Comment (Contrabang) [inline code/game/gamemodes/cult/blood_magic.dm:874]: using get_turf on a turf should always return the same turf
```suggestion
	var/turf/T = get_turf(target)
```

Comment (HMBGERDO) [reply code/game/gamemodes/cult/blood_magic.dm:874]: If you're sure, why not

File: code/game/gamemodes/cult/blood_magic.dm [hunk 2]
```diff
@@ -735,148 +735,163 @@
...
-// This should really be split into multiple procs
-/obj/item/melee/blood_magic/manipulator/afterattack(atom/target, mob/living/carbon/human/user, proximity)
-	if(user.holy_check())
+/obj/item/melee/blood_magic/manipulator/proc/restore_blood(mob/living/carbon/human/user, mob/living/carbon/human/H)
+	if(uses == 0)
 		return
-	if(proximity)
-		if(ishuman(target))
-			var/mob/living/carbon/human/H = target
+	if(!H.dna || (NO_BLOOD in H.dna.species.species_traits) || H.dna.species.exotic_blood != null)
```
Comment (Sirryan2002) [inline code/game/gamemodes/cult/blood_magic.dm:741]:
```suggestion
	if(!H.dna || (NO_BLOOD in H.dna.species.species_traits) || !isnull(H.dna.species.exotic_blood))
```

File: code/game/gamemodes/cult/blood_magic.dm [hunk 3]
```diff
@@ -735,148 +735,163 @@
...
+	var/ratio = uses / overall_damage
+	if(H == user)
+		to_chat(user, "<span class='warning'>Your blood healing is far less efficient when used on yourself!</span>")
+		ratio *= 0.35 // Healing is half as effective if you can't perform a full heal
+		uses -= round(overall_damage) // Healing is 65% more "expensive" even if you can still perform the full heal
+	if(ratio > 1)
+		ratio = 1
+		uses -= round(overall_damage)
+		H.visible_message("<span class='warning'>[H] is fully healed by [H == user ? "[H.p_their()]" : "[H]'s"] blood magic!</span>",
+		"<span class='cultitalic'>You are fully healed by [H == user ? "your" : "[user]'s"] blood magic!</span>")
```
Comment (Sirryan2002) [inline code/game/gamemodes/cult/blood_magic.dm:772]:
```suggestion
			"<span class='cultitalic'>You are fully healed by [H == user ? "your" : "[user]'s"] blood magic!</span>")
```

File: code/game/gamemodes/cult/blood_magic.dm [hunk 4]
```diff
@@ -735,148 +735,163 @@
...
+	else
+		H.visible_message("<span class='warning'>[H] is partially healed by [H == user ? "[H.p_their()]" : "[H]'s"] blood magic.</span>",
+		"<span class='cultitalic'>You are partially healed by [H == user ? "your" : "[user]'s"] blood magic.</span>")
```
Comment (Sirryan2002) [inline code/game/gamemodes/cult/blood_magic.dm:775]:
```suggestion
			"<span class='cultitalic'>You are partially healed by [H == user ? "your" : "[user]'s"] blood magic.</span>")
```

File: code/game/gamemodes/cult/blood_magic.dm [hunk 5]
```diff
@@ -735,148 +735,163 @@
...
+		return
+	var/charge_loss = uses
+	restore_blood(user, H)
+	heal_human_damage(user, H)
+	charge_loss = charge_loss - uses
+	if(!uses)
+		to_chat(user, "<span class='danger'>You use the last of your charges to heal [H == user ? "yourself" : "[H]"], and the spell dissipates!</span>")
+	else
+		to_chat(user, "<span class='cultitalic'>You use [charge_loss] charge\s, and have [uses] remaining.</span>")
+	return
```
Comment (Sirryan2002) [inline code/game/gamemodes/cult/blood_magic.dm:799]:
```suggestion
```
dangling return, it does nothing

File: code/game/gamemodes/cult/blood_magic.dm [hunk 6]
```diff
@@ -735,148 +735,163 @@
...
+	if(uses == 0)
+		return
+	var/missing = M.maxHealth - M.health
+	if(!missing)
+		to_chat(user, "<span class='warning'>[M] doesn't require healing!</span>")
+		return
+	if(uses > missing)
+		M.adjustHealth(-missing)
+		M.visible_message("<span class='warning'>[M] is fully healed by [user]'s blood magic!</span>",
+		"<span class='cultitalic'>You are fully healed by [user]'s blood magic!</span>")
```
Comment (Sirryan2002) [inline code/game/gamemodes/cult/blood_magic.dm:811]:
```suggestion
			"<span class='cultitalic'>You are fully healed by [user]'s blood magic!</span>")
```

File: code/game/gamemodes/cult/blood_magic.dm [hunk 7]
```diff
@@ -735,148 +735,163 @@
...
+		uses -= missing
+	else
+		M.adjustHealth(-uses)
+		M.visible_message("<span class='warning'>[M] is partially healed by [user]'s blood magic!</span>",
+		"<span class='cultitalic'>You are partially healed by [user]'s blood magic.</span>")
```
Comment (Sirryan2002) [inline code/game/gamemodes/cult/blood_magic.dm:816]:
```suggestion
			"<span class='cultitalic'>You are partially healed by [user]'s blood magic.</span>")
```

File: code/game/gamemodes/cult/blood_magic.dm [hunk 8]
```diff
@@ -735,148 +735,163 @@
...
+		return
+	H.blood_volume -= 100
+	uses += 50
+	user.Beam(H, icon_state = "drainbeam", time = 10)
+	playsound(get_turf(H), 'sound/misc/enter_blood.ogg', 50)
+	H.visible_message("<span class='danger'>[user] has drained some of [H]'s blood!</span>",
+					"<span class='userdanger'>[user] has drained some of your blood!</span>")
+	to_chat(user, "<span class='cultitalic'>Your blood rite gains 50 charges from draining [H]'s blood.</span>")
+	new /obj/effect/temp_visual/cult/sparks(get_turf(H))
+	return
```
Comment (Sirryan2002) [inline code/game/gamemodes/cult/blood_magic.dm:842]:
```suggestion
```
dangling return

**Review by Sirryan2002 — APPROVED** [review]
as long as you test this pretty thoroughly in all the use cases you're changing with blood rites it all looks good to me. Nothing immediate pops out to me, potentially some of the `.Beam()` procs could have theirs times switched from deciseconds to seconds but that's really purely a personal preference in this case.

**Review by S34NW — APPROVED** [review]

**Pull request merged**